src/core/ext/xds/xds_client.cc

```diff
@@ -188,15 +193,16 @@ class XdsClient::ChannelState::AdsCallState
       if (state.resource != nullptr) return;
       // Start timer.
       ads_calld_ = std::move(ads_calld);
-      Ref(DEBUG_LOCATION, "timer").release();
-      timer_pending_ = true;
-      grpc_timer_init(
-          &timer_,
-          ExecCtx::Get()->Now() + ads_calld_->xds_client()->request_timeout_,
-          &timer_callback_);
+      timer_handle_ = GetDefaultEventEngine()->RunAfter(
```

Comment:
Because `XdsClient` is a global object, we definitely do not want to use the EE impl used by the calling channel.  That wouldn't make sense semantically, since the same `XdsClient` could be used by two different channels that each use a different EE impl.

I think using `GetDefaultEventEngine()` is fine for now.  If we need a way to override the default just for the `XdsClient` in the future, we can add a special `SetEventEngineFactoryForXdsClient()` API.